fdbserver/workloads/TenantManagementWorkload.actor.cpp

```diff
@@ -622,6 +622,51 @@ struct TenantManagementWorkload : TestWorkload {
 		}
 	}
 
+	// set a random watch on a tenant
+	ACTOR static Future<ErrorOr<Void>> watchTenant(TenantManagementWorkload* self, TenantName tenant) {
+		try {
+			state Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(self->dataDb, tenant));
+			loop {
+				try {
+					state Future<Void> watch = tr->watch(doubleToTestKey(deterministicRandom()->random01()));
+					wait(tr->commit());
+					wait(watch);
+					return Void();
+				} catch (Error& e) {
+					wait(tr->onError(e));
+				}
+			}
+		} catch (Error& e) {
+			return e;
+		}
```

Comment:
We have a generic actor that can handle this. You could remove the try/catch from here and then at the call site use `ErrorOr<Void> result = errorOr(watchTenant(...))`